### PR TITLE
Remove Escape Chars

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: demisto/etc/issues/27200
## Description
removed `_` and `-` from special markdown chars regex to avoid adding a `\` beforehand what could break urls.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by Bargenish** [comment]
We use this function for almost every section of the README (for all entities).
The cases in which Markdown looks at those chars as special ones are when a line begins with a dash or if an underscore surrounds an entire word.
It doesn't happen very often that the yaml file contains such cases. Url's are more common.

**Comment by yaakovi** [comment]
> We use this function for almost every section of the README (for all entities).
> The cases in which Markdown looks at those chars as special ones are when a line begins with a dash or if an underscore surrounds an entire word.
> It doesn't happen very often that the yaml file contains such cases. Url's are more common.

but it does have a use...
need to support both.

**Comment by Bargenish** [comment]
@yaakovi Added support for both cases.

File: demisto_sdk/commands/generate_docs/common.py [hunk 1]
```diff
@@ -170,7 +170,20 @@ def string_escape_md(st, minimal_escaping=False, escape_multiline=False, escape_
         for c in '|':
             st = st.replace(c, '\\' + c)
     else:
-        st = "".join(["\\" + str(c) if c in r"\`*_{}[]()#+-!" else str(c) for c in st])
+        st = "".join(f'\\{str(c)}' if c in r"\`*{}[]()#+!" else str(c) for c in st)
+
+        # The following code adds an escape character for '-' and '_' following cases:
+        # 1. The string begins with a dash. e.g: - This input specifies the entry id
+        # 1. The string has a word which wrapped with an underscore in it . e.g: This input _specifies_ the entry id
```
Comment (yaakovi) [inline demisto_sdk/commands/generate_docs/common.py:177]:
```suggestion
        # 2. The string has a word which wrapped with an underscore in it. e.g: This input _specifies_ the entry id
```

**Review by yaakovi — APPROVED** [review]
update changelog as well.

**Pull request merged**